DashAISoftware/DashAI
# Configure package logging using CLI option

# Summary

This pr aims to provide a mechanism to configure the DashAI subpackages loggers level by using the CLI option `--logging-level`. The levels are the same as the Python logging library. By default, the set level is "INFO"

Note that --logging-level not only affects the DashAI loggers, but also the datasets (which is set to the same level as DashAI) and the SQLAlchemy (which is only activated when logging level is DEBUG).

## Type of change

- Back end new feature.

## Changes

- Add a new `logging_level` option to DashAI CLI, which acts an enum of the available logging levels.
- A new line was added in the application factory (create_app) where the level of all the subpackages logger is set according to the value passed in the CLI.
- Improve datasets module logging.
- Add and improve some docstrings.


## How to Test

Run DashAI with logger level as `DEBUG`, upload a dataset and then, delete the datasets. Check the main and datasets generated logs:


```bash
$ python -m DashAI --logging-level DEBUG
```

Run DashAI with logger level as `WARN`, upload a dataaset and then, delete the datasets. Check the generated logs. No DashAI log should be found.

```bash
$ python -m DashAI --logging-level WARN
```

Run the test:

```bas
$ pytest. 
```


## Notes

This pr depends on #169
